### PR TITLE
add idiom 301 Recursive Fibonacci sequence

### PR DESCRIPTION
**Pull request by simonvonhackewitz**
Not sure if there is a better way to do this. The idiom description says that naive recursion is extremely inefficient for this task, but the other language do it similarly.

### CODE REVIEW

**Comment by simonvonhackewitz** [comment]
I've added a tail recursive version as a second example. I've only included a running example for it and left out the code version, take a look at the html file and let me know if a different way is proffered.

**Comment by simonvonhackewitz** [comment]
Also both version don't work for negative numbers. Does that matter for an idiom?

And if so, whats the best way to handle it? Change the argument type to `u32`, then it will have a different type than the default type of constants or loop variables, or add a precondition?

File: src/idiom301.fuzion [hunk 1]
```diff
@@ -0,0 +1,3 @@
+fib(n i32) i32 =>
+  if n<2 then n
+  else        (fib (n-1)) + (fib (n-2))
```
Comment (michaellilltokiwa) [inline src/idiom301.fuzion:3]: line feed missing at EOF.
in vscode you can use option "files.insertFinalNewline": true
then this is done automagically

Comment (simonvonhackewitz) [reply src/idiom301.fuzion:3]: 🙈
thanks 👍️

**Review by michaellilltokiwa — APPROVED** [review]

**Pull request merged**